README.md

```diff
@@ -42,6 +46,225 @@ nm.init()
 });
 ```
 
-## References
+## HTTP APIs
+
+### Connect to network
+
+POST `/connect-network`
+
+Body:
+
+```JSON
+{
+  "value": {
+    "ssid": "awesome ssid",
+    "passphrase": "supersecret"
+  }
+}
+```
+
+Success Response
+
+```HTTP
+HTTP/1.1 200 OK
+{
+  "ssid": "awesome ssid",
+  "passphrase": "supersecret"
+}
+```
+
+Error 404 Response
+
+```HTTP
+HTTP/1.1 404 Not Found
+{
+  "code": 404,
+  "message": "Could not find nearby AccessPoints with SSID: awesome ssid",
+  "data": {}
+}
+```
+
+Error 500 Response
+
+```HTTP
+HTTP/1.1 500 Internal Server Error
+{
+  "code": 500,
+  "message": "Could not connectNetwork",
+  "data": {...}
+}
+```
+
+### List nearby networks
+
+GET `/list-nearby-networks`
+
+Success Response
+
+```HTTP
+HTTP/1.1 200 OK
+[
+  {
+    "Ssid": "awesome ssid",
+    "Frequency": "5",
+    "security": {
+      "open": false,
+      "encryption": "wpa"
+    }
+  }
+]
+```
+
+Error 500 Response
+
+```HTTP
+HTTP/1.1 500 Internal Server Error
+{
+  "code": 500,
+  "message": "Could not listNearbyNetworks",
+  "data": {...}
+}
+```
+
+### Get current activated network
+
+GET `/current-network`
+
+Success Response
+
+```HTTP
+HTTP/1.1 200 OK
+{
+  "ssid": "awesome ssid"
+}
+```
+
+Error 404 Response
+
+```HTTP
+HTTP/1.1 404 Not Found
+{
+  "code": 404,
+  "message": "You're not currently connected to a wireless network",
+  "data": {}
+}
+```
+
+Error 500 Response
 
-- [NetworkManager DBUS Types](https://developer.gnome.org/NetworkManager/stable/nm-dbus-types.html)
+```HTTP
+HTTP/1.1 500 Internal Server Error
+{
+  "code": 500,
+  "message": "Could not getCurrentNetwork",
+  "data": {...}
+}
+```
+
+### Forget network
+
+POST `/forget-network`
+
+```JSON
+{
+  "value": {
+    "ssid": "awesome ssid"
+  }
+}
+```
+
+Success Response
+
+```HTTP
+HTTP/1.1 200 OK
+{
+  "ssid": "awesome ssid"
+}
+```
+
+Error 404 Response
+
+```HTTP
+HTTP/1.1 404 Not Found
+{
+  "code": 404,
+  "message": "Could not find nearby AccessPoints with SSID: \"awesome ssid\"",
+  "data": {}
+}
+```
+
+Error 500 Response
+
+```HTTP
+HTTP/1.1 500 Internal Server Error
+{
+  "code": 500,
+  "message": "Could not forgetNetwork",
+  "data": {...}
+}
+```
+
+### Toggle WiFi connection
+
+POST `/toggle-wifi`
+
+```JSON
+{
+  "value": true
```

Comment:
Missing description of what this data is here – is this the request or the response body?